Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -272,15 +290,19 @@ static OSStatus outputRenderCallBack(void *inRefCon,
 {
     OCTAudioEngine *myEngine = (__bridge OCTAudioEngine *)inRefCon;
 
-    int bytesToCopy = ioData->mBuffers[0].mDataByteSize;
+    UInt32 bytesToCopy = ioData->mBuffers[0].mDataByteSize;
     SInt16 *targetBuffer = (SInt16 *)ioData->mBuffers[0].mData;
 
     int32_t availableBytes;
-    SInt16 *buffer = TPCircularBufferTail(&myEngine->_buffer, &availableBytes);
-    int32_t sampleCount = MIN(bytesToCopy, availableBytes);
+    SInt16 *buffer = TPCircularBufferTail(&myEngine->_incomingBuffer, &availableBytes);
+    UInt32 sampleCount = MIN(bytesToCopy, availableBytes);
 
+    if (bytesToCopy > availableBytes) {
+        memset(targetBuffer, 0, bytesToCopy);
```

Comment:
I'm wondering why are you filling all targetBuffer with zeroes. Shouldn't we just copy availableBytes and drop leftovers?